### ISSUE TITLE
System.ComponentModel.Win32Exception  when using dotnet-ef tools

### ISSUE THREAD

**Issue by djkiwe** [dotnet/efcore#13163]
Hi, when trying to use any command in dotnet-ef tools installed globally,  I get the following exception:
```
System.ComponentModel.Win32Exception (2): No such file or directory
   at Interop.Sys.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Boolean setUser, UInt32 userId, UInt32 groupId, Int32& lpChildPid, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean shouldThrow)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at Microsoft.EntityFrameworkCore.Tools.Exe.Run(String executable, IReadOnlyList`1 args, String workingDirectory, Boolean interceptOutput)
   at Microsoft.EntityFrameworkCore.Tools.Project.FromFile(String file, String buildExtensionsDir, String framework, String configuration, String runtime)
   at Microsoft.EntityFrameworkCore.Tools.RootCommand.Execute()
   at Microsoft.DotNet.Cli.CommandLine.CommandLineApplication.Execute(String[] args)
   at Microsoft.EntityFrameworkCore.Tools.Program.Main(String[] args)
```

### Steps to reproduce
Tried to list migrations 
```
dotnet ef migrations list
```

### Further technical details
project.csproj
```
<Project Sdk="Microsoft.NET.Sdk.Web">
  <PropertyGroup>
    <TargetFramework>netcoreapp2.1</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="IdentityModel" Version="3.9.0" />
    <PackageReference Include="IdentityServer4" Version="2.2" />
    <PackageReference Include="IdentityServer4.AccessTokenValidation" Version="2.6.0" />
    <PackageReference Include="IdentityServer4.EntityFramework" Version="2.1.1" />
    <PackageReference Include="Microsoft.AspNetCore.All" Version="2.1.3" />
    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="2.1.2" />
    <PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="2.1.1.1" />
    <PackageReference Include="Sendgrid" Version="9.9.0" />
  </ItemGroup>

  <ItemGroup>
    <Folder Include="wwwroot\" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.App" />
  </ItemGroup>
</Project>
```

dotnet --info 
```
.NET Core SDK (reflecting any global.json):
 Version:   2.1.401
 Commit:    91b1c13032

Runtime Environment:
 OS Name:     fedora
 OS Version:  28
 OS Platform: Linux
 RID:         fedora.28-x64
 Base Path:   /usr/lib64/dotnet/sdk/2.1.401/

Host (useful for support):
  Version: 2.1.3
  Commit:  N/A

.NET Core SDKs installed:
  2.1.401 [/usr/lib64/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.NETCore.App 2.1.3 [/usr/lib64/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```
Any ideas?

**Comment by ajcvickers** [comment]
@djkiwe What is the result of running `which dotnet`?

**Comment by ajcvickers** [comment]
**EF Team Triage:** Closing this issue as the requested additional details have not been provided and we have been unable to reproduce it.

*BTW this is a canned response and may have info or details that do not directly apply to this particular issue. While we'd like to spend the time to uniquely address every incoming issue, we get a lot traffic on the EF projects and that is not practical. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we use canned responses for common triage decisions.*

**Comment by djkiwe** [comment]
Sorry for the delayed response. The result of `which dotnet` is `/usr/bin/dotnet`

**Comment by ajcvickers** [comment]
@djkiwe There's nothing obvious we can see wrong here. Can you post the full project so we can attempt to reproduce the behavior you are seeing?

**Comment by djkiwe** [comment]
I'm not sure it's related to this project in particular as I have the same issue with other freshly created projects using the dotnet cli. When I use `dotnet ef` (the one loaded with this line` <DotNetCliToolReference Include="Microsoft.EntityFrameworkCore.Tools.DotNet" Version="2.0.3" />` ), the program works perfectly.
I think it has something to do with running global dotnet cli tools.

**Comment by ajcvickers** [comment]
@djkiwe Can you post the output when running with the '--verbose' flag?

**Comment by djkiwe** [comment]
Update released today using dotnet-sdk 2.1.402 fixed the issue. Now I'm able to use global dotnet-ef tools. I don't know what caused the error in the end. Thanks a lot for your time.
